hw/templates/snax_streamer_wrapper.sv.tpl

```diff
@@ -0,0 +1,159 @@
+  input  logic [ NumInputPorts-1:0][StreamerDataWidth-1:0] acc2stream_data_i,
+  input  logic [ NumInputPorts-1:0]                        acc2stream_valid_i,
+  output logic [ NumInputPorts-1:0]                        acc2stream_ready_o,
+
+  // Ouput ports from accelerator to streamer
+  output logic [NumOutputPorts-1:0][StreamerDataWidth-1:0] stream2acc_data_o,
+  output logic [NumOutputPorts-1:0]                        stream2acc_valid_o,
+  input  logic [NumOutputPorts-1:0]                        stream2acc_ready_i,
+
+  //-----------------------------
+  // TCDM ports
+  //-----------------------------
+  // Request
+  output logic [TCDMReqPorts-1:0]                        tcdm_req_write_o,
+  output logic [TCDMReqPorts-1:0][    TCDMAddrWidth-1:0] tcdm_req_addr_o,
+  //Note that tcdm_req_amo_i is 4 bits based on reqrsp definition
+  output logic [TCDMReqPorts-1:0][                  3:0] tcdm_req_amo_o, 
+  output logic [TCDMReqPorts-1:0][  NarrowDataWidth-1:0] tcdm_req_data_o,
+  //Note that tcdm_req_user_core_id_i is 5 bits based on Snitch definition
+  output logic [TCDMReqPorts-1:0][                  4:0] tcdm_req_user_core_id_o,
+  output logic [TCDMReqPorts-1:0]                        tcdm_req_user_is_core_o,
+  output logic [TCDMReqPorts-1:0][NarrowDataWidth/8-1:0] tcdm_req_strb_o,
+  output logic [TCDMReqPorts-1:0]                        tcdm_req_q_valid_o,
+
+  // Response
+  input  logic [TCDMReqPorts-1:0]                        tcdm_rsp_q_ready_i,
+  input  logic [TCDMReqPorts-1:0]                        tcdm_rsp_p_valid_i,
+  input  logic [TCDMReqPorts-1:0][  NarrowDataWidth-1:0] tcdm_rsp_data_i,
+
+  //-----------------------------
+  // CSR control ports
+  //-----------------------------
+  // Request
+  input  logic [31:0] csr_req_bits_data_i,
+  input  logic [31:0] csr_req_bits_addr_i,
+  input  logic        csr_req_bits_write_i,
+  input  logic        csr_req_valid_i,
+  output logic        csr_req_ready_o,
+
+  // Response
+  input  logic        csr_rsp_ready_i,
+  output logic        csr_rsp_valid_o,
+  output logic [31:0] csr_rsp_bits_data_o
+);
+
+  //-----------------------------
+  // Wiring and combinational logic
+  //-----------------------------
+
+  // Fixed ports that are defaulted
+  // towards the TCDM from the streamer
+  always_comb begin
+    for(int i = 0; i < TCDMReqPorts; i++ ) begin
+      tcdm_req_amo_o          [i] = '0;
+      tcdm_req_user_core_id_o [i] = '0;
+      tcdm_req_user_is_core_o [i] = '0;
+      tcdm_req_strb_o         [i] = '1;
+    end
+  end
+
+  // Streamer module that is generated
+  // with template mechanics
+  StreamerTop i_streamer_top (	
+    //-----------------------------
+    // Clocks and reset
+    //-----------------------------
+    .clock ( clk_i   ),
+    .reset ( ~rst_ni ),
+
+    //-----------------------------
+    // Accelerator ports
+    //-----------------------------
+% for idx, dw in enumerate(cfg["snax_streamer_cfg"]["fifo_writer_params"]["fifo_width"]):
```

Comment:
Bug!! Maybe you mean something like `["fifo_num"]`, which equals to `len(cfg["snax_streamer_cfg"]["fifo_writer_params"]["fifo_width"])`?

Reply:
Fixed!